src/gu/sceGuLight.c

```diff
@@ -10,15 +10,21 @@
 
 void sceGuLight(int light, int type, int components, const ScePspFVector3 *position)
 {
-	GuLightSettings* settings = &light_settings[light];
+	int offset = light * 3;
+	int ctype;
 
-	sendCommandf(settings->xpos,position->x);
-	sendCommandf(settings->ypos,position->y);
-	sendCommandf(settings->zpos,position->z);
+	if (components == GU_POWERED_DIFFUSE) {
+		ctype = GU_DIFFUSE_AND_SPECULAR;
+	} else if (components == GU_DIFFUSE_AND_SPECULAR) {
```

Comment:
```suggestion
	} else if (components == GU_DIFFUSE_AND_SPECULAR || components == 0) {
```
Otherwise the result will be different if the value given is 0.

Reply:
I think that sending 0 should use just `DIFFUSE`

Reply:
The original code was `kind = (components^6) < 1 ? 1 : 0;`,  which results in kind being set to 1 if components is set to 0 or 6. So setting it to DIFFUSE instead of DIFFUSE_AND_SPECULAR in the case of 0 would be an API change. Do you think the original behavior is a bug?

Reply:
To me it looks a bug of our previous implementation…